keras/src/backend/common/remat_scope.py

```diff
@@ -0,0 +1,127 @@
+from keras.src.backend.common import global_state
+
+
+class RematScope:
+    """A context manager for enabling rematerialization in Keras.
+
+    Rematerialization (gradient checkpointing) trades memory for computation by
+    recomputing intermediate activations during the backward pass. This is
+    particularly useful for training large models or large batch sizes within
+    limited memory constraints.
+
+    Args:
+        mode: Rematerialization mode to apply.
+            Options:
+            - "full": Apply rematerialization globally to all supported
+              operations.
+            - "activations": Apply rematerialization only to activation layers.
+            - "larger_than": Apply rematerialization to layers with output sizes
+              larger than `output_size_threshold`.
+            - "list_of_layers": Apply rematerialization to a specific list of
+              layer names.
+            - None: Disable rematerialization.
+        output_size_threshold: Output size threshold for the
+            `"larger_than"` mode. Layers producing outputs larger than this
+            threshold will be rematerialized. Default is `1024`.
+        layer_names: List of layer names for the
+            `"list_of_layers"` mode. Default is an empty list.
+
+    Examples:
+    Using "list_of_layers" mode:
+
+    ```python
+    from keras.src.backend.common.remat_scope import RematScope
```

Comment:
This namespace is private, we can't show it in a code example

Reply:
Just exposed it and updated the example.